deployment/src/main/resources/kubedeployment_editor.groovy

```diff
@@ -0,0 +1,206 @@
+                                if(temp.getString("deploymentname").equals( depmeta.get("name") ) && temp.getString("containername").equals(container.get("name"))) {
+                                    hasSidecarbeenAdded = true
+                                    matchfound = true
+                                    break
+                                }
+                            }
+                            // If a match is found enter the updated container into the list
+                            if (matchfound){
+                                String containerFilepath
+                                String sidecarFilepath
+                                if( loglocations.getJSONObject(i).get("path").toString().startsWith('/') && loglocations.getJSONObject(i).get("path").toString().endsWith('/') ){
+                                    containerFilepath = loglocations.getJSONObject(i).get("path")
+                                    sidecarFilepath = "/opt/tests"+loglocations.getJSONObject(i).get("deploymentname")+"/"+loglocations.getJSONObject(i).get("containername")+loglocations.getJSONObject(i).get("path")
+                                }else if( loglocations.getJSONObject(i).get("path").toString().startsWith('/') && !loglocations.getJSONObject(i).get("path").toString().endsWith('/') ){
+                                    containerFilepath = loglocations.getJSONObject(i).get("path")+"/"
+                                    sidecarFilepath = "/opt/tests"+loglocations.getJSONObject(i).get("deploymentname")+"/"+loglocations.getJSONObject(i).get("containername")+loglocations.getJSONObject(i).get("path")+"/"
+                                }else if( !loglocations.getJSONObject(i).get("path").toString().startsWith('/') && loglocations.getJSONObject(i).get("path").toString().endsWith('/') ){
+                                    containerFilepath = "/"+loglocations.getJSONObject(i).get("path")
+                                    sidecarFilepath = "/opt/tests/"+loglocations.getJSONObject(i).get("deploymentname")+"/"+loglocations.getJSONObject(i).get("containername")+loglocations.getJSONObject(i).get("path")
+                                }else{
+                                    containerFilepath = "/"+loglocations.getJSONObject(i).get("path")+"/"
+                                    sidecarFilepath = "/opt/tests/"+loglocations.getJSONObject(i).get("deploymentname")+"/"+loglocations.getJSONObject(i).get("containername")+loglocations.getJSONObject(i).get("path")+"/"
+                                }
+
+                                // New volume mount for the container
+                                // new volume mount for the sidecar container
+                                // Echo the path into a file in the sidecar container so it knows that it is a log path
+                                Map new_VolumeMount = ["name":"logfilesmount"+logcontainers, "mountPath": containerFilepath ]
```

Comment:
Add design and deployment diagrams to the PR itself in the description, as well as to developer/user guide.